R/principal_summary.R

```diff
@@ -0,0 +1,160 @@
+        stringr::str_detect(.data$pod_name, "Bed Days$") ~ "bed_days"
+      ),
+      change = .data$principal - .data$baseline,
+      change_pcnt = .data$change / .data$baseline
+    ) |>
+    dplyr::arrange(.data$activity_type, .data$measure, .data$pod_name) |>
+    dplyr::select(
+      "pod_name",
+      "activity_type",
+      "baseline",
+      "principal",
+      "change",
+      "change_pcnt"
+    )
+}
+
+
+principal_summary_table <- function(data) {
+  data |>
+    dplyr::mutate(
+      dplyr::across(
+        "principal",
+        \(.x) gt_bar(.x, scales::comma_format(1), "#686f73", "#686f73")
+      ),
+      dplyr::across("change", \(.x) gt_bar(.x, scales::comma_format(1))),
+      dplyr::across("change_pcnt", \(.x) gt_bar(.x, scales::percent_format(1)))
+    ) |>
+    dplyr::mutate(
+      "activity_type" = as.character(.data$activity_type),
+      "activity_type" = dplyr::case_when(
+        # include admissions/beddays in gt groupnames
+        stringr::str_detect(.data$pod_name, "Admission") ~
+          glue::glue("{.data$activity_type} Admissions"),
+        stringr::str_detect(.data$pod_name, "Bed Days") ~
+          glue::glue("{.data$activity_type} Bed Days"),
+        .default = .data$activity_type
+      )
+    ) |>
+    gt::gt(groupname_col = "activity_type") |>
+    gt::cols_align(align = "left", columns = "pod_name") |>
+    gt::cols_label(
+      "pod_name" = "Point of Delivery",
+      "baseline" = "Baseline",
+      "principal" = "Principal",
+      "change" = "Change",
+      "change_pcnt" = "Percent Change"
+    ) |>
+    gt::fmt_integer("baseline") |>
+    gt::cols_width(
+      .data$principal ~ gt::px(150),
+      .data$change ~ gt::px(150),
+      .data$change_pcnt ~ gt::px(150)
+    ) |>
+    gt::cols_align(
+      align = "left",
+      columns = c("baseline", "principal", "change", "change_pcnt")
+    ) |>
+    gt_theme()
+}
+
+
+principal_los_pods <- function() {
+  get_activity_type_pod_measure_options() |>
+    dplyr::filter(.data$activity_type != "aae") |>
+    dplyr::distinct(.data$activity_type, .data$pod, .data$pod_name) |>
+    dplyr::bind_rows(data.frame(
+      activity_type = "aae",
+      pod = "aae",
+      pod_name = "A&E Attendance"
+    )) |>
+    dplyr::mutate(dplyr::across("pod_name", forcats::fct_inorder))
+}
+
+
+get_principal_high_level <- function(r, measures, sites) {
+  r$results$default |>
+    dplyr::filter(.data$measure %in% measures) |>
+    dplyr::select("pod", "sitetret", "baseline", "principal") |>
+    dplyr::mutate(dplyr::across(
+      "pod",
+      ~ ifelse(
+        stringr::str_starts(.x, "aae"),
+        "aae",
+        .x
+      )
+    )) |>
+    dplyr::group_by(.data$pod, .data$sitetret) |>
+    dplyr::summarise(dplyr::across(where(is.numeric), sum), .groups = "drop") |>
+    trust_site_aggregation(sites)
+}
+
+
+get_activity_type_pod_measure_options <- function() {
+  get_golem_config("pod_measures") |>

```

Comment:
We can _probably_ expect repos using {reskit} to have a `golem-config.yaml` file copied from nhp_outputs. So maybe at some point we might need a function to check and error if the config isn't found. Any thoughts on this?